src/autocomplete/autocompleter.ts

```diff
@@ -306,34 +306,38 @@ export class Autocompleter extends Application {
           // Before the searche result we can have one or two specion command:
           //   - Create New (always there)
           //   - Select current Journal (in search journal page only)
-          const resultStartOffset = this._journalSearchFromPageEdition || this._currentMode == AutocompleteMode.journalPageSearch ? 2 : 1;
+          const resultStartOffset = this._initialSearchOffset();
           switch (event.key) {
             case 'Enter': {
               if (this._currentMode===AutocompleteMode.docSearch) {
                 if (this._searchDocType === null) return;
 
                 // if it's 0, pop up the add item dialog
-                if (!this._focusedMenuKey) {
+                if (this._focusedMenuKey===0) {
```

Comment:
I agree with you, I thought you used the other writing elsewhere...

Reply:
I probably did... I've been sloppy about this historically and it's come back to bite me recently when the value is null ;)